src/js/inverted-index.js

```diff
@@ -0,0 +1,197 @@
+/**
+ * InvertedIndex Class
+ * @class
+ */
+class Index {
+
+  /**
+   * class constructor
+   * @constructor
+   */
+  constructor() {
+    this.indexObject = {};
+    this.fileArray = [];
+  }
+
+  /**
+   * validJson
+   *
+   * Reads the JSON data using FileReader
+   *
+   * @param {object} file - Content of the file
+   * @param {string} filename - Name of the file
+   * @return {string}
+   */
+  validJson(filename, file) {
+    if (!file.length) {
+      return 'Empty file';
+    }
+
+    try {
+      const jsonData = JSON.parse(file);
+      if (this.checkTitleAndText(jsonData)) {
+        return 'No title or text';
+      }
+      this.createIndex(filename, jsonData);
```

Comment:
### Critical

It makes more sense if the `validJSON` is called from `createIndex`, so the flow is, when a user tries to create and index, it is implicitly first validated, not the other way around.